buy_together_app/views.py

```diff
@@ -7,3 +9,34 @@ def main_page(request):
 
 def description_page(request):
     return render(request, 'buy_together_app/description.html')
+
+
+def signup_user(request, msg=""):
+    form = SupplierForm()
+    formm = ClientForm()
+    context = {'form': form, 'formm': formm, 'message': [msg]}
+    return render(request, 'buy_together_app/sign_up.html', context)
+
+
+def signup_supplier(request):
+    if request.method == 'POST':
+        form = SupplierForm(request.POST)
+        if form.is_valid():
+            try:
+                form.save()
+                return redirect('Main Page')
+            except Exception:
+                return signup_user(request, 'Username is taken')
+        return signup_user(request, 'Request is invalid')
+
+
+def signup_client(request):
+    if request.method == 'POST':
+        form = ClientForm(request.POST)
+        if form.is_valid():
+            try:
+                form.save()
+                return redirect('Main Page')
+            except Exception:
+                return signup_user(request, 'Username is taken')
+        return signup_user(request, 'Request is invalid')
```

Comment:
The two functions are almost identical. Can we do something like this instead?
```suggestion
def signup(request, form_type):
    if request.method == 'POST':
        form = form_type(request.POST)
        if form.is_valid():
            try:
                form.save()
                return redirect('Main Page')
            except Exception:
                return signup_user(request, 'Username is taken')
        return signup_user(request, 'Request is invalid')
```
You can have `signup_client` and `signup_supplier` wrap this function and provide this input:
```python
def signup_client(request):
    return signup(request, ClientForm)
```

Reply:
Done!